gapic/samplegen/samplegen.py

```diff
@@ -511,3 +512,59 @@ def generate_sample(sample,
                                                 imports=[],
                                                 calling_form=calling_form,
                                                 calling_form_enum=utils.CallingForm)
+
+
+def generate_manifest(fpaths_and_samples,
```

Comment:
Reference implementation: [sample-tester/gen_manifest](https://github.com/googleapis/sample-tester/blob/master/gen_manifest/gen_manifest.py#L76).

In particular, to emit YAML anchors I chose to emit the YAML by hand, partly because it wasn't obvious that PyYaml could do it. I just found a [link](https://www.wenbiancheng.com/question/yaml/18065427-Generating%20anchors%20with%20PyYAML.dump()?) that suggests that it can, though I'd want the anchor name to be human-readable.

Reply:
I'm in the middle of that judgement call, balancing with the competing forces of "generating yaml by hand is deeply tedious and unrewarding" vs "fighting with PyYaml makes me want to cry tears of frustration because of all the boilerplate".
There's no _good_ solution, just a matter of which one sounds less unpleasant.

Reply:
Wound up rolling my own yaml emitter.

Reply:
Nice!